tests/init/pos/early-promote-simple.scala

```diff
@@ -0,0 +1,4 @@
+class P() {
+    val a = 1
+    List(this)
+}
```

Comment:
It would be good to add a test that a warm inner class object promotes to hot because of the outer object being hot.